### PR TITLE
Print RPM messages to the user

### PR DESCRIPTION
**Pull request by m-blaha**
The RPM messages contain vital information for debugging RPM transaction.
Currently are these messages only written to the dnf5.log file. This patch
also prints them to the user during the transaction execution.

Resolves: https://github.com/rpm-software-management/dnf5/issues/1710
Related: https://bugzilla.redhat.com/show_bug.cgi?id=2312906

### CODE REVIEW

**Review by evan-goode — APPROVED** [review]
Looks great overall!

File: libdnf5/rpm/transaction.hpp [hunk 1]
```diff
@@ -321,6 +321,10 @@ class Transaction {
     /// @param file_path  new file path
     void set_script_out_file(const std::string & file_path);
 
+    /// Retrieve recently emitted rpm log messages. After the method is called,
+    /// the stored rpm log messages are cleared.
+    std::vector<std::string> get_rpm_messages();
```
[the remark this answers is not in the document]
Comment (pmatilai) [reply libdnf5/rpm/transaction.hpp:326]: Maybe "pop" or "dequeue" would make the behavior clearer (the latter is annoying to type :sweat_smile: )

[the remark this answers is not in the document]
Comment (m-blaha) [reply libdnf5/rpm/transaction.hpp:326]: Thanks guys, I'll go with `extract` which is consistent with what `RpmLogGuard` has.

**Comment by evan-goode** [comment]
The F39 tests are failing in the CI due to https://bugzilla.redhat.com/show_bug.cgi?id=2315859, but they pass locally (after rebasing to main). Merging.

**Pull request merged**